src/platform_impl/macos/window.rs

```diff
@@ -1339,11 +1340,14 @@ impl WindowExtMacOS for WinitWindow {
             self.set_style_mask_sync(new_mask);
             shared_state_lock.is_simple_fullscreen = false;
 
-            if let Some(presentation_opts) = shared_state_lock.save_presentation_opts {
+            let save_presentation_opts = shared_state_lock.save_presentation_opts;
+            let frame = shared_state_lock.saved_standard_frame();
+            drop(shared_state_lock);
```

Comment:
when I first added the `windowDidChangeScreen` listener I ran into an issue where we need to acquire the window shared state mutex to check `is_simple_fullscreen`. But calling `app.setPresentationOptions` (which we also do in the above branch / when toggling simple fullscreen on or off) that results in the `windowDidChangeScreen` listener being called synchronously on the same (main) thread.

I wasn't sure the best way to handle this but there were a few other places in this file where the shared state lock is explicitly dropped like this, presumably for similar reasons.